pandas/core/arrays/categorical.py

```diff
@@ -2312,6 +2313,22 @@ def replace(self, to_replace, value, inplace: bool = False):
         if not inplace:
             return cat
 
+    # ------------------------------------------------------------------------
+    # String methods interface
+    def _str_map(self, f, na_value=np.nan, dtype=np.dtype(object)):
```

Comment:
why do these need to be here?

Reply:
or rather if you explain / show *how* these are called on the appropriate object

Reply:
@TomAugspurger I would maybe add a comment / docstring here to explain this is overriding the method of `ObjectStringArrayMixin` to ensure the string functions are only applied on the categories and not on the densified values

Reply:
yep thanks

Reply:
the reason i am asking for this is that our accessor dispatch is already a bit opaque and this is going once step further